assembly.py

```diff
@@ -339,6 +340,55 @@ def parser_assemble_trinity(parser=argparse.ArgumentParser()):
 __commands__.append(('assemble_trinity', parser_assemble_trinity))
 
 
+def assemble_spades(
+    inBam,
+    outFasta,
+    spades_opts='',
+    previously_assembled_contigs=None,
+    mem_limit_gb=4,
+    threads=1,
+):
+    ''' De novo RNA-seq assembly with the SPAdes assembler.
+
+    Inputs:
+        inBam - reads to assemble.  May include both paired and unpaired reads.
+        previously_assembled_contigs - (optional) already-assembled contigs from the same sample.
+
+    Outputs:
+        outFasta - the assembled contigs.  Note that, since this is RNA-seq assembly, for each assembled genomic region there may be
+            several contigs representing different variants of that region.
+
+    Params:
+        mem_limit_gb - max memory to use
+        threads - number of threads to use
+        spades_opts - (advanced) custom command-line options to pass to the SPAdes assembler
+
+    '''
+
+    with tools.picard.SamToFastqTool().execute_tmp(inBam, includeUnpaired=True) as (reads_fwd, reads_bwd, reads_unpaired):
+        try:
+            tools.spades.SpadesTool().assemble(reads_fwd=reads_fwd, reads_bwd=reads_bwd, reads_unpaired=reads_unpaired,
+                                               contigs_untrusted=previously_assembled_contigs,
+                                               contigs_out=outFasta, spades_opts=spades_opts, mem_limit_gb=mem_limit_gb,
+                                               threads=threads)
+        except subprocess.CalledProcessError as e:
+            raise DenovoAssemblyError('SPAdes assembler failed: ' + str(e))
```

Comment:
Is the SPAdes error message helpful enough to identify if the failure is due to insufficient input size? If not, maybe the `DenovoAssemblyError ` should include some input metrics.